Common/TableProducer/mcCollsExtra.cxx

```diff
@@ -135,7 +135,7 @@ struct mcCollisionExtra {
       if (iter != sortedIndices.end()) {
         int index = iter - sortedIndices.begin();
         for (int iMcColl = index + 1; iMcColl < index + 17; iMcColl++) {
```

Comment:
Declaring `iMcCol1` as `size_t` here is probably better.

Reply:
But `index + 1` is not guaranteed to be non-negative, is it?

Reply:
`index` (which should be obtained with `std::distance`, probably) cannot be negative since it should not be before the `.begin()`.

Reply:
Yes, but it has type `int`.

Reply:
This is why I recommend to use `auto index = std::distance(iter, sortedIndices.begin());`